src/view.rs

```diff
@@ -37,101 +36,109 @@ pub fn render_game(app: &mut ApplicationState, ctx: &mut Context) -> GameResult<
     if !world.player.is_dead {
         graphics::set_color(ctx, color::RED)?;
-        render_player(&world.player, ctx)?;
+        render_player(ctx, &world.player)?;
     }
 
     Ok(())
 }
 
 /// Renders all the stars in the background
-fn render_stars(world: &World, ctx: &mut Context) -> GameResult<()> {
-    let img = graphics::Image::solid(ctx, 1, color::GREY)?;
-    let mut sb = SpriteBatch::new(img);
+fn render_stars(ctx: &mut Context, world: &World, resources: &Resources) -> GameResult<()> {
+    let mut spritebatch = SpriteBatch::new(resources.star_image.clone());
 
     // Iterate through the stars list and draw them with a rotation based on their index in the
     // list - this isn't a truly random rotation, but it works visually
     for (i, star) in world.stars.iter().enumerate() {
-        sb.add(graphics::DrawParam {
+        let scale = 0.05 * (star.size as f32);
+        spritebatch.add(graphics::DrawParam {
             dest: Point2::new(star.x() as f32, star.y() as f32),
             rotation: (i as f32 / 100.0) * 2.0 * std::f32::consts::PI,
-            scale: Point2::new(star.size as f32, star.size as f32),
+            scale: Point2::new(scale, scale),
             .. Default::default()
         });
     }
 
-    graphics::draw_ex(ctx, &sb, graphics::DrawParam { ..Default::default() })
+    graphics::draw_ex(ctx, &spritebatch, graphics::DrawParam { ..Default::default() })
 }
 
-/// Renders a particle
-pub fn render_particle(particle: &Particle, ctx: &mut Context) -> GameResult<()> {
-    let radius = 5.0 * particle.ttl as f32;
-    let pt = Point2::new(particle.x() as f32, particle.y() as f32);
-    graphics::circle(ctx, DrawMode::Fill, pt, radius, 2.0)
+/// Renders all the particles
+pub fn render_particles(ctx: &mut Context, world: &World, resources: &Resources) -> GameResult<()> {
+    let mut spritebatch = SpriteBatch::new(resources.circle_image.clone());
+    for particle in &world.particles {
+        let scale = (0.4 * particle.ttl) as f32;
+        spritebatch.add(graphics::DrawParam {
+            dest: Point2::new(particle.x() as f32, particle.y() as f32),
+            offset: Point2::new(0.5, 0.5),
+            scale: Point2::new(scale, scale),
+            ..Default::default()
+        });
+    }
+    graphics::draw_ex(ctx, &spritebatch, graphics::DrawParam { ..Default::default() })
 }
 
 /// Renders a bullet
-pub fn render_bullet(bullet: &Bullet, ctx: &mut Context) -> GameResult<()> {
+pub fn render_bullet(ctx: &mut Context, bullet: &Bullet) -> GameResult<()> {
     let pt = Point2::new(bullet.x() as f32, bullet.y() as f32);
     graphics::circle(ctx, DrawMode::Fill, pt, bullet.radius() as f32, 2.0)
 }
 
 /// Renders an enemy
-pub fn render_enemy(enemy: &Enemy, ctx: &mut Context) -> GameResult<()> {
+pub fn render_enemy(ctx: &mut Context, enemy: &Enemy) -> GameResult<()> {
     let pt = Point2::new(enemy.x() as f32, enemy.y() as f32);
     graphics::circle(ctx, DrawMode::Fill, pt, enemy.radius() as f32, 0.5)
```

Comment:
Since we are using `SpriteBatch`, we could use it to draw the enemies as well, right? They are also circles...